app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -683,26 +680,44 @@ void loadFromIntent(final Intent intent) {
             }
         }
 
-        Uri uri = intent.getData();
-
         boolean openInWindow = false;
         boolean openInBackground = false;
+        boolean openInKioskMode = false;
+
+        Uri dataUri = intent.getData();
+        Uri targetUri = null;
+        Bundle extras;
+
+        if (dataUri != null && dataUri.getScheme().equals("wolvic") && dataUri.getHost().equals("com.igalia.wolvic")) {
+            Log.d(LOGTAG, "Parsing custom URI from intent: " + dataUri);
+
+            extras = new Bundle();
+            Set<String> keys = dataUri.getQueryParameterNames();
+            for (String key : keys) {
+                if (key.equals("url") || key.equals("homepage"))
+                    extras.putString(key, dataUri.getQueryParameter(key));
+                else
+                    extras.putBoolean(key, Boolean.parseBoolean(dataUri.getQueryParameter(key)));
+            }
+        } else {
+            targetUri = intent.getData();
+            extras = intent.getExtras();
+        }
 
-        Bundle extras = intent.getExtras();
         if (extras != null) {
             // If there is no data uri and there is a url parameter we get that
-            if (uri == null && extras.containsKey("url")) {
-                uri = Uri.parse(extras.getString("url"));
+            if (extras.containsKey("url")) {
```

Comment:
Either the comment is outdated or you miss a null check

Reply:
Thanks. I have rewritten the comment and added a null check.